File: backend/src/main/java/org/thepaddlers/controller/PaymentsController.java
package org.thepaddlers.controller;

import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;
import org.thepaddlers.service.StripeService;
import java.util.Map;

@RestController
@RequestMapping("/api/payments")
public class PaymentsController {
    private final StripeService stripeService;

    public PaymentsController(StripeService stripeService) {
        this.stripeService = stripeService;
    }

    @PostMapping("/create-intent")
    public ResponseEntity<?> createIntent(@RequestBody Map<String, Object> body) {
        // Expect amount, currency, userId, etc. in body
        try {
            String clientSecret = stripeService.createPaymentIntent(body);
            return ResponseEntity.ok(Map.of("client_secret", clientSecret));
        } catch (Exception e) {
            return ResponseEntity.status(500).body(Map.of("error", e.getMessage()));
        }
    }
}

File: services/backend/src/main/kotlin/com/thepaddlers/payments/PaymentsController.kt
package com.thepaddlers.payments

import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.*
import org.springframework.beans.factory.annotation.Autowired

@RestController
@RequestMapping("/payments")
class PaymentsController @Autowired constructor(
    private val stripePaymentService: StripePaymentService
) {
    @PostMapping("/create-intent")
    fun createIntent(
        @RequestHeader(value = "Idempotency-Key", required = false) idempotencyKey: String?,
        @RequestBody req: CreateIntentRequest
    ): ResponseEntity<CreateIntentResponse> {
        val paymentIntent = stripePaymentService.createPaymentIntent(
            req.amount,
            req.currency,
            mapOf("reservationId" to req.reservationId),
            idempotencyKey
        )
        return ResponseEntity.ok(
            CreateIntentResponse(
                clientSecret = paymentIntent.clientSecret,
                paymentIntentId = paymentIntent.id
            )
        )
    }
}

data class CreateIntentRequest(
    val amount: Long,
    val currency: String,
    val reservationId: String
)
data class CreateIntentResponse(
    val clientSecret: String?,
    val paymentIntentId: String?
)

